File: 김광현/week11/42586_기능개발.py
# 먼저 각 작업이 얼마나 시간이 걸리는지를 lst에 저장하기
def solution(progresses, speeds):
    n = len(progresses)
    time = [0] * n
    for i in range(n):
        j = 0
        while progresses[i] < 100:
            progresses[i] += speeds[i]
            j += 1
        if i != 0 and time[i-1] > j:
            time[i] = time[i-1]
        else:
            time[i] = j
            
    answer = []
    i = 0
    j = 1
    while i < (n - 1):
        if time[i] == time[i + 1]:
            i += 1
            j += 1
        else:
            answer.append(j)
            i += 1
            j = 1
    answer.append(j)
    
    return answer

File: 백태균/week11/42586_기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        List<Integer> answer = new ArrayList<>();
        int[] remainProgresses = new int[progresses.length];
        
        for (int i=0; i<progresses.length; i++) {
            remainProgresses[i] = (int) Math.ceil((double) (100 - progresses[i]) / speeds[i]);
        }
        
        int start = remainProgresses[0];
        int cnt = 1;
        for (int i=1; i<=remainProgresses.length; i++) {
            
            if (i == remainProgresses.length) {
                answer.add(cnt);
                break;
            }
            
            if (start >= remainProgresses[i]) {
                cnt++;
            } else {
                answer.add(cnt);
                start = remainProgresses[i];
                cnt = 1;
            }
        }
        
        return answer.stream().mapToInt(x -> x).toArray();
    }
}